p2p/simulations/events.go

```diff
@@ -28,13 +28,16 @@ func NewEvent(v interface{}) *Event {
 	switch v := v.(type) {
 	case *Node:
 		event.Type = EventTypeNode
-		event.Node = v
+		node := *v
```

Comment:
We need to copy the object so the event represents the state of the object at this time (not copying means the object can be mutated in the future and the event is then different, e.g. `node.Up=true`).

I have added a comment: https://github.com/ethersphere/go-ethereum/blob/d7eb6b5a70d4f49887ce982068007821dad8b407/p2p/simulations/events.go#L50-L51